Assign to self before returning in init methods

LumioseSil opened the pull request:
#### 4c8ed9ea14550faa4067caf7e5985a71ee569760
<pre>
Assign to self before returning in init methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=255341">https://bugs.webkit.org/show_bug.cgi?id=255341</a>

Reviewed by Alexey Proskuryakov.

ARC requires this and this is good practice anyway,
even if these files are not being compiled with ARC.

Yes, there were some exceptions simply because reordering
statements in legacy code is too risky/not worth the effort.

* Source/WebCore/PAL/pal/system/mac/WebPanel.mm:
  (-[WebPanel init]):
* Source/WebKit/Shared/Cocoa/WKNSDictionary.mm:
  (-[WKNSDictionary initWithObjects:forKeys:count:]):
* Source/WebKit/UIProcess/Cocoa/WKReloadFrameErrorRecoveryAttempter.mm:
  (-[WKReloadFrameErrorRecoveryAttempter initWithCoder:]):
* Source/WebKit/UIProcess/ios/forms/WKDatePickerViewController.mm:
  (-[WKDatePickerViewController initWithDelegate:]):
* Source/WebKit/UIProcess/ios/forms/WKDateTimeInputControl.mm:
  (-[WKDateTimeInputControl initWithView:]):
* Source/WebKit/UIProcess/ios/forms/WKFormColorControl.mm:
  (-[WKFormColorControl initWithView:]):
* Source/WebKit/UIProcess/ios/forms/WKFormSelectControl.mm:
  (-[WKFormSelectControl initWithView:]):
* Source/WebKit/UIProcess/ios/forms/WKSelectMenuListViewController.mm:
  (-[WKSelectMenuListViewController initWithDelegate:]):
* Source/WebKit/UIProcess/ios/forms/WKTimePickerViewController.mm:
  (-[WKTimePickerViewController initWithDelegate:]):
* Source/WebKitLegacy/mac/Misc/WebDownload.mm:
  (-[WebDownload init:]):
* Source/WebKitLegacy/mac/Misc/WebIconDatabase.mm:
  (-[WebIconDatabase init]):
* Source/WebKitLegacy/mac/WebView/WebResource.mm:
  (-[WebResourcePrivate init]):
* Tools/DumpRenderTree/mac/DumpRenderTreeDraggingInfo.mm:
  (-[DumpRenderTreeDraggingInfo initWithImage:offset:pasteboard:source:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
  (-[MockContentFilterEnabler initWithCoder:]):
* Tools/TestWebKitAPI/ios/DragAndDropSimulatorIOS.mm:
  (-[MockDropSession initWithProviders:location:window:allowMove:]):
  (-[MockDragSession initWithWindow:allowMove:]):
* Tools/WebKitTestRunner/mac/WebKitTestRunnerWindow.mm:
  (-[WebKitTestRunnerWindow initWithContentRect:styleMask:backing:defer:]):

Canonical link: <a href="https://commits.webkit.org/262971@main">https://commits.webkit.org/262971@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/661a9ebd6d04a6aed4092f859a2147edb78849b2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3125 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3185 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3295 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4537 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3485 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3093 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3246 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3212 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2715 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3156 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3476 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2801 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4342 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/974 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2777 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2621 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2748 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2828 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4078 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3200 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2552 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2786 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2779 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/777 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2777 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3044 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/UIProcess/ios/forms/WKDateTimeInputControl.mm:
```diff
@@ -341,11 +341,11 @@ - (instancetype)initWithView:(WKContentView *)view
         break;
     default:
-        [self release];
         return nil;
```
Why is this change OK?

AZero13 replied:
Because it is in an init method, this would be an overrelease.

aproskuryakov replied:
Please elaborate. What else releases the object?

AZero13 replied:
It is never initialized in the first place is what I mean, so releasing it makes no sense.

AZero13 replied:
Actually I was wrong. This should stay. @aproskuryakov

aproskuryakov commented on Source/WebKitLegacy/mac/Misc/WebDownload.mm:
```diff
@@ -225,12 +225,14 @@ - (void)_setRealDelegate:(id)delegate
+        return nil;
+
+    // _webInternal can be set up before init by _setRealDelegate
+    if (_webInternal == nil) {
```
As you are touching these lines anyway, please remove braces around the single line block.

AZero13 replied:
Addressed!

aproskuryakov reviewed (commented):
I'm not much of an Objective-C expert, but this seems like an improvement.

LumioseSil committed:
Assign to self before returning in init methods
https://bugs.webkit.org/show_bug.cgi?id=255341

Reviewed by Alexey Proskuryakov.

ARC requires this and this is good practice anyway,
even if these files are not being compiled with ARC.

Yes, there were some exceptions simply because reordering
statements in legacy code is too risky/not worth the effort.

* Source/WebCore/PAL/pal/system/mac/WebPanel.mm:
  (-[WebPanel init]):
* Source/WebKit/Shared/Cocoa/WKNSDictionary.mm:
  (-[WKNSDictionary initWithObjects:forKeys:count:]):
* Source/WebKit/UIProcess/Cocoa/WKReloadFrameErrorRecoveryAttempter.mm:
  (-[WKReloadFrameErrorRecoveryAttempter initWithCoder:]):
* Source/WebKit/UIProcess/ios/forms/WKDatePickerViewController.mm:
  (-[WKDatePickerViewController initWithDelegate:]):
* Source/WebKit/UIProcess/ios/forms/WKDateTimeInputControl.mm:
  (-[WKDateTimeInputControl initWithView:]):
* Source/WebKit/UIProcess/ios/forms/WKFormColorControl.mm:
  (-[WKFormColorControl initWithView:]):
* Source/WebKit/UIProcess/ios/forms/WKFormSelectControl.mm:
  (-[WKFormSelectControl initWithView:]):
* Source/WebKit/UIProcess/ios/forms/WKSelectMenuListViewController.mm:
  (-[WKSelectMenuListViewController initWithDelegate:]):
* Source/WebKit/UIProcess/ios/forms/WKTimePickerViewController.mm:
  (-[WKTimePickerViewController initWithDelegate:]):
* Source/WebKitLegacy/mac/Misc/WebDownload.mm:
  (-[WebDownload init:]):
* Source/WebKitLegacy/mac/Misc/WebIconDatabase.mm:
  (-[WebIconDatabase init]):
* Source/WebKitLegacy/mac/WebView/WebResource.mm:
  (-[WebResourcePrivate init]):
* Tools/DumpRenderTree/mac/DumpRenderTreeDraggingInfo.mm:
  (-[DumpRenderTreeDraggingInfo initWithImage:offset:pasteboard:source:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
  (-[MockContentFilterEnabler initWithCoder:]):
* Tools/TestWebKitAPI/ios/DragAndDropSimulatorIOS.mm:
  (-[MockDropSession initWithProviders:location:window:allowMove:]):
  (-[MockDragSession initWithWindow:allowMove:]):
* Tools/WebKitTestRunner/mac/WebKitTestRunnerWindow.mm:
  (-[WebKitTestRunnerWindow initWithContentRect:styleMask:backing:defer:]):

Canonical link: https://commits.webkit.org/262971@main
